### PR TITLE
Resolved Task #1

### PR DESCRIPTION
**Pull request by wladengine**

### CODE REVIEW

File: AsyncAwait.Task1.CancellationTokens/Program.cs [hunk 1]
```diff
@@ -17,45 +19,52 @@ internal class Program
...
+        AskN();
 
         var input = Console.ReadLine();
         while (input.Trim().ToUpper() != "Q")
         {
+            CancellationTokenSource cancellationTokenSource = new();
             if (int.TryParse(input, out var n))
             {
-                CalculateSum(n);
+                CalculateSum(n, cancellationTokenSource.Token);
```
Comment (ASemykinBrl) [inline AsyncAwait.Task1.CancellationTokens/Program.cs]: Please add task await is some appropriate place. It's a bad practice to leave tasks unawaited

File: AsyncAwait.Task1.CancellationTokens/Calculator.cs [hunk 2]
```diff
@@ -1,21 +1,27 @@
 ﻿using System.Threading;
+using System.Threading.Tasks;
 
 namespace AsyncAwait.Task1.CancellationTokens;
 
 internal static class Calculator
 {
     // todo: change this method to support cancellation token
-    public static long Calculate(int n /*, CancellationToken token*/)
+    public static Task<long> CalculateAsync(int n , CancellationToken token)
```
Comment (ASemykinBrl) [inline AsyncAwait.Task1.CancellationTokens/Calculator.cs]: Please make this method async in an appropriate manner

**Review by ASemykinBrl — CHANGES_REQUESTED** [review]
Please review comments

File: AsyncAwait.Task1.CancellationTokens/Program.cs (hunk 1 above)
Comment (wladengine) [reply AsyncAwait.Task1.CancellationTokens/Program.cs]: Fixed

File: AsyncAwait.Task1.CancellationTokens/Calculator.cs [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
 
 internal static class Calculator
 {
     // todo: change this method to support cancellation token
-    public static long Calculate(int n /*, CancellationToken token*/)
-    {
-        long sum = 0;
-
-        for (var i = 0; i < n; i++)
+    public static Task<long> CalculateAsync(int n , CancellationToken token) =>
```
Comment (ASemykinBrl) [inline AsyncAwait.Task1.CancellationTokens/Calculator.cs]: async Task

File: AsyncAwait.Task1.CancellationTokens/Calculator.cs [hunk 4]
```diff
@@ -1,21 +1,25 @@
...
-        }
+            long sum = 0;
+
+            for (var i = 0; i < n; i++)
+            {
+                token.ThrowIfCancellationRequested();
+
+                // i + 1 is to allow 2147483647 (Max(Int32)) 
+                sum += (i + 1);
+                Thread.Sleep(10);
```
Comment (ASemykinBrl) [inline AsyncAwait.Task1.CancellationTokens/Calculator.cs]: Please use async replacement for Thread.Sleep

File: AsyncAwait.Task1.CancellationTokens/Calculator.cs (hunk 3 above)
Comment (wladengine) [reply AsyncAwait.Task1.CancellationTokens/Calculator.cs]: fixed

File: AsyncAwait.Task1.CancellationTokens/Calculator.cs (hunk 4 above)
Comment (wladengine) [reply AsyncAwait.Task1.CancellationTokens/Calculator.cs]: Changed to `await Task.Delay(10)`

**Review by ASemykinBrl — APPROVED** [review]

**Pull request merged**